src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -366,6 +416,26 @@ void AbstractLQPNode::_add_output_pointer(const std::shared_ptr<AbstractLQPNode>
   _outputs.emplace_back(output);
 }
 
+std::shared_ptr<LQPUniqueConstraints> AbstractLQPNode::_forward_unique_constraints() const {
+  Assert(left_input(), "Cannot forward unique constraints from an empty input node.");
+  const auto& input_unique_constraints = left_input()->unique_constraints();
+
+  if constexpr (HYRISE_DEBUG) {
+    // Check whether output expressions are missing
+    const auto& expressions = this->output_expressions();
+    ExpressionUnorderedSet output_expressions{expressions.cbegin(), expressions.cend()};
+    for (const auto& unique_constraint : *input_unique_constraints) {
+      for (const auto& expression : unique_constraint.expressions) {
+        Assert(output_expressions.contains(expression),
+               "Forwarding of constraints is illegal because node misses "
+               "output expressions.");
+      }
```

Comment:
Added a helper func:

```cpp
if constexpr (HYRISE_DEBUG) {
    // Check whether output expressions are missing
    for (const auto& unique_constraint : *input_unique_constraints) {
        Assert(expressions_are_subset_of_output_expressions(unique_constraint.expressions),
               "Forwarding of constraints is illegal because node misses "
               "output expressions.");
    }
  }
```

Reply:
`output_contains_expressions`?

Reply:
Ah, didn't see that the code reads `has_output_expressions`, not `expressions_are_subset_of_output_expressions`. Works for me.